internal/command/meta_backend.go

```diff
@@ -1447,15 +1475,280 @@ func (m *Meta) backend_C_r_S_changed(c *configs.Backend, cHash int, sMgr *clista
+						c.Type,
+						c.Provider.Name,
+						c.ProviderAddr,
+						ws,
+						ws,
+					),
+				))
+				return nil, diags
+
+			case ws == backend.DefaultStateName:
+				// TODO: do we want to prompt for input here (m.Input()), or create automatically unless -readonly flag present?
+				// input := m.UIInput()
+				// desc := fmt.Sprintf("Terraform will create the %q workspace via %q.\n"+
+				// 	"Only 'yes' will be accepted to approve.", backend.DefaultStateName, c.Type)
+				// v, err := input.Input(context.Background(), &terraform.InputOpts{
+				// 	Id:          "approve",
+				// 	Query:       fmt.Sprintf("Workspace %q does not exit, would you like to create one?", backend.DefaultStateName),
+				// 	Description: desc,
+				// })
+				// if err != nil {
+				// 	diags = diags.Append(fmt.Errorf("Failed to confirm default workspace creation: %w", err))
+				// 	return nil, diags
+				// }
+				// if v != "yes" {
+				// 	diags = diags.Append(errors.New("Failed to create default workspace"))
+				// 	return nil, diags
+				// }
+
+				// TODO: Confirm if defaulting to creation on first use (rather than error) is a good idea
+				// Make the default workspace. All other workspaces are user-created via the workspace commands.
+				m.createDefaultWorkspace(c, b)
+			default:
+				diags = diags.Append(err)
+				return nil, diags
+			}
+			// TODO: handle if input is not enabled
+		}
+	}
+	if diags.HasErrors() {
+		return nil, diags
+	}
+
+	// Update backend state file
+	if err := backendSMgr.WriteState(s); err != nil {
+		diags = diags.Append(fmt.Errorf(errBackendWriteSaved, err))
+		return nil, diags
+	}
+	if err := backendSMgr.PersistState(); err != nil {
+		diags = diags.Append(fmt.Errorf(errBackendWriteSaved, err))
+		return nil, diags
+	}
+
+	return b, diags
+}
+
+// createDefaultWorkspace receives a backend made using a pluggable state store, and details about that store's config,
+// and persists an empty state file in the default workspace. By creating this artifact we ensure that the default
+// workspace is created and usable by Terraform in later operations.
+func (m *Meta) createDefaultWorkspace(c *configs.StateStore, b backend.Backend) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+
+	defaultSMgr, err := b.StateMgr(backend.DefaultStateName)
+	if err != nil {
+		diags = diags.Append(fmt.Errorf("Failed to create a state manager for state store %q in  provider %s (%q). This is a bug in Terraform and should be reported: %w",
+			c.Type,
+			c.Provider.Name,
+			c.ProviderAddr,
+			err))
+		return diags
+	}
+	emptyState := states.NewState()
+	if err := defaultSMgr.WriteState(emptyState); err != nil {
+		diags = diags.Append(fmt.Errorf(errStateStoreWorkspaceCreate, c.Type, err))
+		return diags
+	}
+	// TODO - implement Read/Write state RPC methods
+	if err := defaultSMgr.PersistState(nil); err != nil {
+		diags = diags.Append(fmt.Errorf(errStateStoreWorkspaceCreate, c.Type, err))
+		return diags
+	}
```

Comment:
Currently this PR is blocked here - we need to implement the Read/Write state RPC methods